mos_tests/functions/os_cli.py

```diff
@@ -65,19 +65,36 @@ def __call__(self, action, flags='', params='', prefix='', fail_ok=False,
 class OpenStack(CLICLient):
     command = 'openstack'
 
-    def details(self, output):
+    def details(self, output, mapping=('Field', 'Value')):
+        """List with one dict with data"""
         data = json.loads(output)
         if isinstance(data, list):
-            data = {x['Field']: x['Value'] for x in data}
+            data = {x[mapping[0]]: x[mapping[1]] for x in data}
         return data
 
+    def listing(self, output, mapping=False):
+        """List with several dicts with data"""
+        data = json.loads(output)
+        if mapping:
+            if isinstance(data, list):
+                data = [{x[mapping[0]]: x[mapping[1]]} for x in data]
+        return data
+
+    def project_list(self):
+        output = self('project list -f json')
+        return self.listing(output, ('Name', 'ID'))
+
     def project_create(self, name):
         output = self('project create', params='{} -f json'.format(name))
         return self.details(output)
 
     def project_delete(self, name):
         return self('project delete', params=name)
 
+    def user_list(self):
+        output = self('user list -f json')
+        return self.listing(output, ('Name', 'ID'))
```

Comment:
Correct, it will return:
```
[{u'admin': u'80a38a57872c480d8b27047a4ad1883c'},
 {u'glance': u'3e82fe98a1814d239aeb3f67b3cbb805'},
 {u'nova': u'c5b1b55799ae481690a03c4b7a020a82'},
 {u'neutron': u'3e0dfae1bf494a55a4ae839e1640e93d'},
 {u'cinder': u'193628586699409e862e990562672ccc'},
 {u'heat': u'a8ccf9bd47e34d7e9a05126babca28f1'},
 {u'heat-cfn': u'0417c081c8764b71bfe7eac591f13fef'},
 {u'fuel_stats_user': u'21677dcf0ec142e3884e7f7126936e8b'}]
```